### PR TITLE
Fix bug of listing vms && add contract Id.

### PR DESCRIPTION
**Pull request by AlaaElattar**
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1291
- Fix bug of deploying solution and find it in failed deployments while it also appears in list.
- Add contracts of failed deployments.

image

Screenshot from 2023-10-31 11-00-33

### CODE REVIEW

File: packages/playground/src/components/k8s_deployment_table.vue [hunk 1]
```diff
@@ -15,17 +15,20 @@
...
-            </li>
-          </ul>
+          <li v-for="deployment in failedDeployments" :key="deployment.name">
+            {{
+              deployment.nodes.length > 0
+                ? `${deployment.name} on node${deployment.nodes.length > 1 ? "s" : ""}: ${deployment.nodes.join(", ")}`
+                : deployment.name
+            }}
+            with contract id:
+            <template v-if="deployment.contracts && deployment.contracts.length > 0">
```
Comment (0oM4R) [inline packages/playground/src/components/k8s_deployment_table.vue:25]: if there is no contracts, the text will be `with contract id: No contracts found.` ?

File: packages/playground/src/components/k8s_deployment_table.vue [hunk 2]
```diff
@@ -15,17 +15,20 @@
...
+              <span v-for="contract in deployment.contracts" :key="contract.contract_id">
+                {{ contract.contract_id }}&nbsp;
+              </span>
+            </template>
+            <template v-else> No contracts found. </template>
```
Comment (zaelgohary) [inline packages/playground/src/components/k8s_deployment_table.vue:30]: Is this a valid case?

File: packages/playground/src/components/k8s_deployment_table.vue (hunk 1 above)
Comment (AlaaElattar) [reply packages/playground/src/components/k8s_deployment_table.vue:25]: - u're right, it doesn't make sense

File: packages/playground/src/components/k8s_deployment_table.vue (hunk 2 above)
Comment (AlaaElattar) [reply packages/playground/src/components/k8s_deployment_table.vue:30]: I believe it's not .

File: packages/playground/src/components/k8s_deployment_table.vue [hunk 3]
```diff
@@ -15,17 +15,19 @@
...
+          <li v-for="deployment in failedDeployments" :key="deployment.name">
+            {{
+              deployment.nodes.length > 0
+                ? `${deployment.name} on node${deployment.nodes.length > 1 ? "s" : ""}: ${deployment.nodes.join(", ")}`
+                : deployment.name
+            }}
+            with contract id:
+            <template v-if="deployment.contracts && deployment.contracts.length > 0">
+              <span v-for="contract in deployment.contracts" :key="contract.contract_id">
+                {{ contract.contract_id }}&nbsp;
```
Comment (0oM4R) [inline packages/playground/src/components/k8s_deployment_table.vue:27]: typo ?

File: packages/playground/src/components/k8s_deployment_table.vue (hunk 1 above)
Comment (0oM4R) [reply packages/playground/src/components/k8s_deployment_table.vue:25]: i guess we need to set the `with contract id` inside the if

**Comment by zaelgohary** [comment]
There is one case where we won't be able to get the contract id which is if encryption didn't happen. In this case, we need to show a more meaningful message like "unknown contract id as it was not decrypted".

**Comment by AlaaElattar** [comment]
> There is one case where we won't be able to get the contract id which is if encryption didn't happen. In this case, we need to show a more meaningful message like "unknown contract id as it was not decrypted".

will it be good to show msg next to deployment name or just not to show `with contract` as we do with nodeId.

**Comment by zaelgohary** [comment]
> > There is one case where we won't be able to get the contract id which is if encryption didn't happen. In this case, we need to show a more meaningful message like "unknown contract id as it was not decrypted".
> 
> will it be good to show msg next to deployment name or just not to show `with contract` as we do with nodeId.

Sure we can do that.

**Comment by AlaaElattar** [comment]
> > > There is one case where we won't be able to get the contract id which is if encryption didn't happen. In this case, we need to show a more meaningful message like "unknown contract id as it was not decrypted".
> > 
> > 
> > will it be good to show msg next to deployment name or just not to show `with contract` as we do with nodeId.
> 
> Sure we can do that.

done

File: packages/playground/src/components/k8s_deployment_table.vue (hunk 3 above)
Comment (AlaaElattar) [reply packages/playground/src/components/k8s_deployment_table.vue]: I removed it .

**Review by zaelgohary — APPROVED** [review]

**Review by MohamedElmdary — APPROVED** [review]

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**